starter-code/routes/passportRouter.js

```diff
@@ -1,22 +1,76 @@
-const express        = require("express");
-const router         = express.Router();
+const express = require("express");
+const router = express.Router();
 // User model
-const User           = require("../models/user");
+const User = require("../models/user");
 // Bcrypt to encrypt passwords
-const bcrypt         = require("bcrypt");
-const bcryptSalt     = 10;
+const bcrypt = require("bcrypt");
+const bcryptSalt = 10;
 const ensureLogin = require("connect-ensure-login");
-const passport      = require("passport");
-
-
+const passport = require("passport");
 
 router.get("/private-page", ensureLogin.ensureLoggedIn(), (req, res) => {
   res.render("passport/private", { user: req.user });
 });
 
+/* _____ LOGIN __________ */
+
+router.get("/login", (req, res) => {
+  res.render("passport/login");
+});
+
+router.post(
+  "/login",
+  passport.authenticate("local", {
+    successRedirect: "/passport/private-page",
+    failureRedirect: "/login",
+    failureFlash: true,
```

Comment:
You are setting "failureFlash" to true but you are not passing any error to the view (see previous comment).